### PR TITLE
[Docs] updating GenAI 

### PR DESCRIPTION
**Pull request by jillnogold**

### CODE REVIEW

File: docs/genai/data-mgmt/vector-databases.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+Vector databases work by storing vectors that represent the context of a request. These vectors can be generated using various techniques, such as word embeddings. When a request is received, the vector database retrieves the vectors that represent the context of the request and passes them to the model for inference. This allows the model to take into account the context of the request and provide a more accurate response.
+
+In MLRun, you can use vector databases to enrich the context of a request before passing it to a model for inference. This allows you to build more sophisticated models that take into account the context of the request and provide more accurate responses.
```
Comment (theSaarco) [inline docs/genai/data-mgmt/vector-databases.md:8]: Isn't this pretty much repeating the previous sentence?

File: docs/genai/data-mgmt/vector-databases.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+                   collection-name: str):
+    
+    df = vector_db_data.as_df()
+
+    # Create chroma client
+    chroma_client = chromadb.PersistentClient(path=cache_dir)
+    
+    # Add data to the collection    
+    if collection_name in [c.name for c in chroma_client.list_collections()]:
+        chroma_client.delete_collection(name=collection_name)
```
Comment (theSaarco) [inline docs/genai/data-mgmt/vector-databases.md:29]: The comment says `Add data` but this actually deletes the existing collections... Should move the comment to after this initial stage.

File: docs/genai/data-mgmt/vector-databases.md [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    
+    context.logger.info("Vector DB was created")
+```
+
+Then, during inference, you might have a function that retrieves the documents of a specific topic. For example:
+
+```python
+collection = chroma_client.get_collection(collection_name)
+collection.query(query_texts=[topic], n_results=10)
+q_context = " ".join([f"#{str(i)}" for i in results["documents"][0]])
```
Comment (theSaarco) [inline docs/genai/data-mgmt/vector-databases.md:46]: You use `results` here, but this wasn't defined. I assume the previous line should be `results = collection.query...`?

File: docs/genai/data-mgmt/vector-databases.md [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+prompt_template = f"Relevant context: {q_context}\n\n The user's question: {question}"
+
+```
+
+## Supported vector databases
+
+MLRun does not limit the choice of vector databases you can use. You can use any vector database that fits your use case. Some popular vector databases include:
```
Comment (theSaarco) [inline docs/genai/data-mgmt/vector-databases.md:53]: Don't we want to also note that MLRun doesn't come with a VectorDB OOTB, and you need to install your choice of DB? Maybe earlier in this section?

File: docs/genai/deployment/genai_serving.md [hunk 5]
```diff
@@ -0,0 +1,190 @@
...
+```
+
+Next, you can add a model to the function using this code:
+
+```python
+
+hugging_face_serving.add_model(
+    'mymodel',
+    class_name='HuggingFaceModelServer',
+    model_path='123',  # This is not used, just for enabling the process.
```
Comment (theSaarco) [inline docs/genai/deployment/genai_serving.md:24]: I'm not sure what this comment even means. If it's not used, is it mandatory?

File: docs/genai/deployment/genai_serving_graph.md [hunk 6]
```diff
@@ -0,0 +1,109 @@
+(genai-serving-graph)=
+# Gen AI realtime serving graph
+
+During inference, it is common to serve a gen AI model as part of a larger pipeline that includes data preprocessing, model execution, and post-processing. This can be done with MLRun using the real-time serving pipeline feature. Prior to model inference, the context is typically enriched using a vector database, then the input is transformed to input tokens, and finally the model is executed. Pre-processing and post-processing may also include guardrails to ensure the input is valid (for example, prevent the user from asking questions that attempt to exploit the model) as well as output processing, to verify the model does not hallucinate or include data that must not be shared.
+
+## A basic graph
+
+To run a model as part of a larger pipeline, you can use the `set_topology` method of the serving function. The following code shows how to set up a simple pipeline that includes a single step. This example is taken from the [Interactive bot demo using LLMs and MLRun](https://github.com/mlrun/demo-llm-bot) which calls OpenAI ChatGPT model:
```
Comment (theSaarco) [inline docs/genai/deployment/genai_serving_graph.md:8]: Maybe point to the `set_topology` documentation? Also, `set_topology` is only used in the next code snip, after defining the step. Maybe move the "you can use the `set_topology` method of the serving function" note there?

File: docs/genai/deployment/genai_serving_graph.md [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+
+```python
+generate_kwargs = {"max_length": 150, "temperature": 0.9, "top_p": 0.5, "top_k": 25, "repetition_penalty": 1.0}
+response = serving_function.invoke(path='/predict', body={"prompt": "What is MLRun?", **generate_kwargs})
+print(response["outputs"])
+```
+
+## Distributed pipelines
+
+By default, all steps of the serving graph run on the same pod in sequence. It is possible to run different steps on different pods using {ref}`distributed pipelines<distributed-graph>`.Typically you run steps that require CPU on one pod, and steps that require a GPU on a different pod.
```
Comment (theSaarco) [inline docs/genai/deployment/genai_serving_graph.md:109]: "...on a different pod that is running on a potentially different node which has GPU support"

File: docs/genai/deployment/gpu_utilization.md [hunk 8]
```diff
@@ -0,0 +1,42 @@
+(gpu-utilization)=
+# GPU utilization
```
Comment (theSaarco) [inline docs/genai/deployment/gpu_utilization.md:2]: I think this page is important and gives a very good list of options, but is somewhat theoretic - I would expect some examples of how to do things, or at least links to other places where these demos exist. A lot of these techniques are utilized in the GenAI tutorials that we have - we can point there for references.

File: docs/genai/deployment/guardrails-deploy.md [hunk 9]
```diff
@@ -0,0 +1,11 @@
+(guardrails-deploy)=
+# Guardrails for deployment
```
Comment (theSaarco) [inline docs/genai/deployment/guardrails-deploy.md:2]: Same here - this is a theoretic checklist without any references to MLRun-related implementation.

File: docs/genai/genai-flow.md [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+MLRun automates the flow of tuning, validation, and optimizing LLMs to specific data - efficiently over elastic resources. 
+It rapidly deploys scalable real-time serving and application pipelines that host the LLM as well as the data integration and business logic.
+MLRun has built-in monitoring for LLM data, training, model, and resources, with automated model retraining.
+It is an open solution, supporting various LLMs and flexible deployment options (any cloud, and on-prem).
+
+The general stages of developing and scaling a GenAI model across an enterprise entails:
+1. Personalize a pre-trained model with your pre-trained data.
+2. Implement measures to reduce risk: data quality (preparing data before the training), testing (ensure the model is 
+doing exactly what it should be doing), guard  (answers are correct, not toxic, etc.), human feedback (examine it and return the model).
+2. Build a scalable, automated, and continuous development environment.
```
Comment (theSaarco) [inline docs/genai/genai-flow.md:14]:
```suggestion
3. Build a scalable, automated, and continuous development environment.
```
(and next line is `4.`)

**Review by theSaarco — COMMENTED** [review]
Didn't go over the notebooks, just the markdown files.

File: docs/genai/deployment/genai_serving.md (hunk 5 above)
Comment (gilad-shaham) [reply docs/genai/deployment/genai_serving.md:24]: Unfortunately yes, I opened a ticket on that

File: docs/genai/deployment/gpu_utilization.md (hunk 8 above)
Comment (gilad-shaham) [reply docs/genai/deployment/gpu_utilization.md:2]: Agree, we can do that next iteration

File: docs/genai/deployment/guardrails-deploy.md (hunk 9 above)
Comment (gilad-shaham) [reply docs/genai/deployment/guardrails-deploy.md:2]: We may want to leave this out given that there's no content

File: docs/genai/data-mgmt/vector-databases.md (hunk 3 above)
Comment (gilad-shaham) [reply docs/genai/data-mgmt/vector-databases.md:46]: I took a snippet from the tutorial in an early stage, we can take the new snippet and should also reference the tutorial

File: docs/genai/genai-flow.md (hunk 10 above)
Comment (jillnogold) [reply docs/genai/genai-flow.md:14]: After 1., you can use all 2.s. It numbers correctly when publishing

File: docs/genai/data-mgmt/vector-databases.md (hunk 1 above)
Comment (gilad-shaham) [reply docs/genai/data-mgmt/vector-databases.md:8]: Agree, @jillnogold see comment

File: docs/genai/data-mgmt/vector-databases.md (hunk 3 above)
Comment (jillnogold) [reply docs/genai/data-mgmt/vector-databases.md:46]: @gilad - I added the results line as Saar suggested. Noting the rest as a future fix

File: docs/genai/data-mgmt/vector-databases.md (hunk 1 above)
Comment (jillnogold) [reply docs/genai/data-mgmt/vector-databases.md:8]: next version

File: docs/genai/deployment/guardrails-deploy.md (hunk 9 above)
Comment (jillnogold) [reply docs/genai/deployment/guardrails-deploy.md:2]: next time

**Pull request closed without merging**